Keep older versions of cmdliner and ocp-indent

romildo opened the pull request:
###### Motivation for this change

PR https://github.com/NixOS/nixpkgs/pull/26195 (which upgraded `cmdliner` to version 1.0.0) introduced an issue because some ocaml packages are not yet compatible with it.

This PR keeps both versions of `cmdliner` (the old 0.9.8 and the new 1.0.0) and makes the old one the default.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @vbgl

romildo committed:
cmdliner_1_0: init at 1.0.0

Version 1.0 breaks API and some packages does currently not compile
with it. So it is added as a new package.

romildo committed:
cmdliner: 1.0.0 -> 0.9.8

Downgrade to version 0.9.8 because the new version 1.0.0 breaks API and
some pacakges do not yet compile with it.

romildo committed:
ocp-indent: depends on cmdliner minimum version 1.0.0

vbgl commented:
Thanks for this prompt fix.

`ocp-index` seems to be broken as well; I think it’s also related to the `ocp-indent` update.

romildo commented:
> `ocp-index` seems to be broken as well; I think it’s also related to the  `ocp-indent` update.

There is a related issue on upstream: https://github.com/OCamlPro/ocp-index/issues/102

romildo commented:
Should the old version of `ocp-indent` be kept too?

romildo committed:
ocp-indent: readd old version 1.5.2

Some packages (ocp-index, for instance) are not yet compatible with
newer versions of ocp-indent. So version 1.5.2 is still needed.

romildo committed:
ocp-index: add explicitly dependency on ocpIndent_1_5_2

ocp-index does not yet compile with newer versions of ocp-indent.

romildo commented:
I have added `ocp-indent-1.5.2` (attribute `ocpIndent_1_5_2`) so that `ocp-index` can be built successfully. Currently `ocp-index` does not compile with newer versions of `ocp-indent`.
